pkg/util/namer/namer.go

```diff
@@ -37,6 +37,30 @@ func GetPodName(base, suffix string) string {
 	return GetName(base, suffix, util.DNS1123SubdomainMaxLength)
 }
 
+// LimitLength returns a string of at most maxLength characters based on name.
+// It returns name unchanged if it is short enough, otherwise it returns a
+// prefix of name plus a suffix computed from a hash of name.
+func LimitLength(name string, maxLength int) string {
```

Comment:
I'd rather update GetName to handle an empty suffix than have a separate function